src/service/tags/tags_repository.py

```diff
@@ -97,3 +99,57 @@ def get_tags(self, offset: int = 0, max_count: int = None) -> list:
         except Exception as error:
             self.logger.error(error)
             return []
+
+    def add_company_tag_query(self, tag_id: str, companies: list) -> str:
+        query = ""
+        for company_id in companies:
+            if company_id and company_id.strip():
+                company_tag_id = str(uuid.uuid4())
+                query += """
+                    INSERT INTO {table}
+                    VALUES ('{id}', '{tag_id}', '{company_id}');
+                """.format(
+                    table=TableNames.COMPANY_TAG,
+                    id=company_tag_id,
+                    tag_id=tag_id,
+                    company_id=company_id,
+                )
+        return query
+
+    def add_tag_query(self, tag_id: str, tag_name: str) -> str:
+        return """
+            INSERT INTO {table_name}
+            VALUES('{id}', '{name}');
+            """.format(
+            table_name=TableNames.TAG, id=tag_id, name=tag_name
+        )
+
+    def add_tag(self, tag: dict) -> dict:
+        try:
+            tag_id = str(uuid.uuid4())
+            tag_name = tag.get("name")
+            companies = tag.get("companies", [])
+            query = self.add_tag_query(tag_id, tag_name)
+
+            if companies and len(companies) > 0:
```

Comment:
You should verify the "invalid" or "wrong" scenario first.

Reply:
With that change, you should move line 132 `query = self.add_tag_query(tag_id, tag_name)` after that.
You should also validate tag name is a string and isn't empty.